pkg/server/sotw/v3/watches.go

```diff
@@ -0,0 +1,75 @@
+package sotw
+
+import (
+	"context"
+	"reflect"
+
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+)
+
+// watches for all xDS resource types
+type watches struct {
+	responders map[string]*watch
+
+	// indexes is a list of indexes for each dynamic select case which match to a watch
+	cases []reflect.SelectCase
+}
+
+// newWatches creates and initializes watches.
+func newWatches() watches {
+	// deltaMuxedResponses needs a buffer to release go-routines populating it
+	return watches{
+		responders: make(map[string]*watch, int(types.UnknownType)),
+		cases:      make([]reflect.SelectCase, 2), // We use 2 for the default computation here: ctx.Done() + reqCh.Recv()
```

Comment:
Don't we want this to be properly initialized? I assume we'll call RecompueWatches before this ever get read, so maybe just call it directly in newWatches or avoid this logic since it doesn't matter?